api/mesh/v1alpha1/dataplane_helpers.go

```diff
@@ -127,6 +127,26 @@ func (i OutboundInterface) String() string {
 		strconv.FormatUint(uint64(i.DataplanePort), 10))
 }
 
+func NonBackendRefFilter(outbound *Dataplane_Networking_Outbound) bool {
```

Comment:
Any plans for this API that we don't just have `GetNonBackendRefOutbounds`, `GetAllOutbounds` instead? Fine either way I suppose

Reply:
I feel like we may need to filter out external services outbound in the future. Maybe I'm wrong. I'll leave it as is for this PR.